File: app/src/main/java/com/bitflaker/lucidsourcekit/database/daos/DreamTypeDao.java
package com.bitflaker.lucidsourcekit.database.daos;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.bitflaker.lucidsourcekit.database.entities.DreamType;

import java.util.List;

import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Single;

@Dao
public interface DreamTypeDao {
    @Query("SELECT * FROM DreamType")
    Single<List<DreamType>> getAll();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    Completable insertAll(DreamType... types);

    @Delete
    Completable delete(DreamType type);
}

File: app/src/main/java/com/bitflaker/lucidsourcekit/database/dreamjournal/daos/DreamTypeDao.kt
package com.bitflaker.lucidsourcekit.database.dreamjournal.daos

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.bitflaker.lucidsourcekit.database.dreamjournal.entities.DreamType

@Dao
interface DreamTypeDao {
    @Query("SELECT * FROM DreamType")
    suspend fun getAll(): MutableList<DreamType>

    @Insert(onConflict = OnConflictStrategy.Companion.IGNORE)
    suspend fun insertAll(types: Array<DreamType>)

    @Delete
    suspend fun delete(type: DreamType)
}
